app/main/views/jobs.py

```diff
@@ -161,6 +174,30 @@ def cancel_job(service_id, job_id):
     return redirect(url_for('main.service_dashboard', service_id=service_id))
 
 
+@main.route("/services/<service_id>/jobs/<uuid:job_id>/cancel", methods=['GET', 'POST'])
+@login_required
+@user_has_permissions()
+def cancel_letter_job(service_id, job_id):
+    if request.method == 'POST':
+        job = job_api_client.get_job(service_id, job_id)['data']
+        notifications = notification_api_client.get_notifications_for_service(
+            job['service'], job['id']
+        )['notifications']
+        if job['job_status'] != 'finished' or len(notifications) < job['notification_count']:
+            flash("We are still processing these letters, please try again in a minute.", 'try again')
+            return view_job(service_id, job_id)
+        try:
+            number_of_letters = job_api_client.cancel_letter_job(current_service.id, job_id)
+        except HTTPError as e:
+            flash(e.message, 'dangerous')
+            return redirect(url_for('main.view_job', service_id=service_id, job_id=job_id))
+        flash(" {} letters have been cancelled succesfully".format(number_of_letters), 'default_with_tick')
```

Comment:
This will comma format the count of letters, and remove the slight contradiction between ‘cancel’ and ‘success’.
```suggestion
        flash("Cancelled {:,.0f} letters from {}".format(number_of_letters, job['original_file_name']), 'default_with_tick')
```